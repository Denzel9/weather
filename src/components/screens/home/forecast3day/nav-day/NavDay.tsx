import { FunctionComponent } from 'react'
import { IDataInfo, IOneDay } from '../../../../../types/data.interface'
import classNames from 'classnames'

const NavDay: FunctionComponent<{
  data: IDataInfo
  date: IOneDay
  setDate(obj: IOneDay): void
}> = ({ data, date, setDate }) => {
  return (
    <div className=" flex items-center justify-between h-16">
      {data?.weather?.forecast?.forecastday?.map((item) => {
        return (
          <div
            data-testid="divClick"
            key={item.date}
            className="transition-transform hover:scale-110 cursor-pointer"
            onClick={() => setDate({ dayWeather: item, city: data?.oneDays?.city })}
          >
            <img
              data-testid="icon"
              className={classNames(
                date?.dayWeather?.date.slice(8, 10) === item?.date.slice(8, 10)
                  ? 'w-[50px] duration-300'
                  : 'duration-300',
                'w-[30px] m-auto'
              )}
              src={item?.day?.condition?.icon}
              alt={item.date}
            />
            <span className="text-xs">{item.date}</span>
          </div>
        )
      })}
    </div>
  )
}

export default NavDay
