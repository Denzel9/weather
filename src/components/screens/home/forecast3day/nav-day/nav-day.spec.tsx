import { fireEvent, render, screen } from '@testing-library/react'
import { IOneDay } from '../../../../../types/data.interface'
import '@testing-library/jest-dom'
import NavDay from './NavDay'
import { QueryClient } from 'react-query'
import { QueryClientProvider } from 'react-query'
import renderer from 'react-test-renderer'

jest.mock('axios', () => ({
  __esModule: true,

  default: {
    get: () => ({
      data: { id: 1, name: 'den' },
    }),
  },
}))

const setDate = jest.fn()

const date: IOneDay = {
  city: 'Moscow',
  dayWeather: {
    date: '2023-10-27',
    day: {
      mintemp_c: 10,
      maxtemp_c: 20,
      maxwind_kph: 6,
      daily_chance_of_rain: 50,
      uv: 1,
    },
    hour: [
      {
        uv: 1,
        chance_of_rain: 30,
        wind_kph: 12,
        feelslike_c: 15,
        temp_c: 12,
        time: ' 2023-10-27 00:00',
      },
    ],
  },
}

const data = {
  chart: [
    {
      name: '00:00',
      uv: '-2',
    },
    {
      name: '00:00',
      uv: '-2',
    },
    {
      name: '00:00',
      uv: '-2',
    },
  ],
  oneDays: {
    city: 'Moscow',
    dayWeather: {
      date: '2023-10-27',
      day: {
        mintemp_c: 10,
        maxtemp_c: 20,
        maxwind_kph: 6,
        daily_chance_of_rain: 50,
        uv: 1,
      },
      hour: [
        {
          uv: 1,
          chance_of_rain: 30,
          wind_kph: 12,
          feelslike_c: 15,
          temp_c: 12,
          time: ' 2023-10-27 00:00',
        },
      ],
    },
  },
  weather: {
    current: {
      condition: {
        icon: '//cdn.weatherapi.com/weather/64x64/day/368.png',
        text: 'Небольшой снег',
      },
      temp_c: 12,
    },
    forecast: {
      forecastday: [
        {
          day: { mintemp_c: 10, maxtemp_c: 20, maxwind_kph: 6, daily_chance_of_rain: 50, uv: 1 },
          date: '2023-10-25',
          hour: [
            {
              uv: 2,
              chance_of_rain: 50,
              wind_kph: 5,
              feelslike_c: 12,
              temp_c: 2,
              time: '2023-10-27 00:00',
            },
          ],
        },
        {
          day: { mintemp_c: 10, maxtemp_c: 20, maxwind_kph: 6, daily_chance_of_rain: 50, uv: 1 },
          date: '2023-10-26',
          hour: [
            {
              uv: 2,
              chance_of_rain: 50,
              wind_kph: 5,
              feelslike_c: 12,
              temp_c: 2,
              time: '2023-10-27 00:00',
            },
          ],
        },
        {
          day: { mintemp_c: 10, maxtemp_c: 20, maxwind_kph: 6, daily_chance_of_rain: 50, uv: 1 },
          date: '2023-10-27',
          hour: [
            {
              uv: 2,
              chance_of_rain: 50,
              wind_kph: 5,
              feelslike_c: 12,
              temp_c: 2,
              time: '2023-10-27 00:00',
            },
          ],
        },
      ],
    },
    location: {
      name: '"Москва"',
      localtime: '2023-10-27 14:47',
      temp_c: 12,
    },
  },
}

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
})

describe('navigation day', () => {
  it('render', () => {
    render(
      <QueryClientProvider client={queryClient}>
        <NavDay data={data as any} date={date!} setDate={setDate} />
      </QueryClientProvider>
    )

    expect(screen.getByText(/2023-10-25/i)).toBeInTheDocument()
    expect(screen.getByText(/2023-10-26/i)).toBeInTheDocument()
    expect(screen.getByText(/2023-10-27/i)).toBeInTheDocument()
  })

  it('change day', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <NavDay data={data as any} date={date!} setDate={setDate} />
      </QueryClientProvider>
    )

    expect(screen.queryByAltText(/2023-10-27/i)).toHaveClass('w-[50px]')
    fireEvent.click(screen.queryByAltText(/2023-10-26/i)!)
    expect(setDate).toBeCalled()
    expect(await screen.findByAltText(/2023-10-27/i)).toHaveClass('w-[50px]')
  })

  it('snapshot', async () => {
    const component = renderer
      .create(
        <QueryClientProvider client={queryClient}>
          <NavDay data={data as any} date={date!} setDate={setDate} />
        </QueryClientProvider>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })
})
