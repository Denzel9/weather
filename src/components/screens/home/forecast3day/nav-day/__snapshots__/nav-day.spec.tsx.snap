// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation day snapshot 1`] = `
<div
  className=" flex items-center justify-between h-16"
>
  <div
    className="transition-transform hover:scale-110 cursor-pointer"
    data-testid="divClick"
    onClick={[Function]}
  >
    <img
      alt="2023-10-25"
      className="duration-300 w-[30px] m-auto"
      data-testid="icon"
    />
    <span
      className="text-xs"
    >
      2023-10-25
    </span>
  </div>
  <div
    className="transition-transform hover:scale-110 cursor-pointer"
    data-testid="divClick"
    onClick={[Function]}
  >
    <img
      alt="2023-10-26"
      className="duration-300 w-[30px] m-auto"
      data-testid="icon"
    />
    <span
      className="text-xs"
    >
      2023-10-26
    </span>
  </div>
  <div
    className="transition-transform hover:scale-110 cursor-pointer"
    data-testid="divClick"
    onClick={[Function]}
  >
    <img
      alt="2023-10-27"
      className="w-[50px] duration-300 w-[30px] m-auto"
      data-testid="icon"
    />
    <span
      className="text-xs"
    >
      2023-10-27
    </span>
  </div>
</div>
`;
